src/roiextractors/extractors/numpyextractors/numpyextractors.py

```diff
@@ -97,209 +104,282 @@ class NumpySegmentationExtractor(SegmentationExtractor):
 
     def __init__(
         self,
-        image_masks,
-        raw=None,
-        dff=None,
-        deconvolved=None,
-        neuropil=None,
-        accepted_lst=None,
-        mean_image=None,
-        correlation_image=None,
-        roi_ids=None,
-        roi_locations=None,
-        sampling_frequency=None,
-        rejected_list=None,
-        channel_names=None,
-        movie_dims=None,
+        image_masks: Union[PathType, np.ndarray],
+        roi_response_traces: dict[str, Union[PathType, np.ndarray]],
+        sampling_frequency: FloatType,
+        roi_ids: Optional[list] = None,
+        accepted_roi_ids: Optional[list] = None,
+        rejected_roi_ids: Optional[list] = None,
+        roi_locations: Optional[ArrayType] = None,
+        summary_images: Optional[dict[str, Union[PathType, np.ndarray]]] = None,
+        background_image_masks: Optional[Union[PathType, np.ndarray]] = None,
+        background_response_traces: Optional[dict[str, Union[PathType, np.ndarray]]] = None,
+        background_ids: Optional[list] = None,
     ):
-        """Create a NumpySegmentationExtractor from a .npy file.
+        """Create a NumpySegmentationExtractor from a set of .npy files or a set of np.ndarrays.
 
         Parameters
         ----------
-        image_masks: np.ndarray
-            Binary image for each of the regions of interest
-        raw: np.ndarray
-            Fluorescence response of each of the ROI in time
-        dff: np.ndarray
-            DfOverF response of each of the ROI in time
-        deconvolved: np.ndarray
-            deconvolved response of each of the ROI in time
-        neuropil: np.ndarray
-            neuropil response of each of the ROI in time
-        mean_image: np.ndarray
-            Mean image
-        correlation_image: np.ndarray
-            correlation image
-        roi_ids: int list
-            Unique ids of the ROIs if any
-        roi_locations: np.ndarray
-            x and y location representative of ROI mask
-        sampling_frequency: float
-            Frame rate of the movie
-        rejected_list: list
-            list of ROI ids that are rejected manually or via automated rejection
-        channel_names: list
-            list of strings representing channel names
-        movie_dims: tuple
-            height x width of the movie
-        """
-        SegmentationExtractor.__init__(self)
-        if isinstance(image_masks, (str, Path)):
-            image_masks = Path(image_masks)
-            if image_masks.is_file():
-                assert image_masks.suffix == ".npy", "'image_masks' file is not a numpy file (.npy)"
+        image_masks: Union[PathType, np.ndarray]
+            Binary image for each of the regions of interest.
```

Comment:
The description doesn't seem to indicate what type of arg would take the Path type